### PR TITLE
[SPARK-44967][SQL][CONNECT] Unit should be considered first before using Boolean for TreeNodeTag

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
Currently, there are a lot of `TreeNodeTag[Boolean]` are defined.
In fact, developers don't require the boolean value boxed into `TreeNodeTag`, just want know it as a flag.

### Why are the changes needed?
Unit should be considered first before using Boolean for TreeNodeTag.

### Does this PR introduce _any_ user-facing change?
'No'.
Just update inner implementation.

### How was this patch tested?
N/A

### Was this patch authored or co-authored using generative AI tooling?
'No'.

### CODE REVIEW

**Comment by beliefer** [comment]
ping @cloud-fan

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala [hunk 1]
```diff
@@ -53,6 +53,8 @@ import org.apache.spark.util.collection.BitSet
 private class MutableInt(var i: Int)
 
 // A tag of a `TreeNode`, which defines name and type
+// Note: In general, developers only care about its tagging capabilities,
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala:56]:
```suggestion
// Note: In general, if developers only care about its tagging capabilities,
```

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala [hunk 2]
```diff
@@ -53,6 +53,8 @@ import org.apache.spark.util.collection.BitSet
...
+// so Unit should be considered first before using Boolean.
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala:57]:
```suggestion
// then Unit should be considered first before using Boolean.
```

**Review by cloud-fan — APPROVED** [review]

**Review by yaooqinn — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**